radis/lbl/broadening.py

```diff
@@ -2820,6 +2844,33 @@ def rough_sum_on_grid():
         line2grid_projection_left,
     )
 
+
+def get_molar_mass(df):
+
+    molpar = MolParams()
+
+    if "id" in df.columns:
+        id = df.id.unique()
+        id = id[0]
+
+    else:
+        id = df.attrs["id"]
+
+    if "iso" in df.columns:
+        iso_set = df.iso.unique()
+        molar_mass = {}
+        for iso in iso_set:
+            molar_mass[iso] = molpar.get(id, iso, "mol_mass")
+
+        req = df["iso"].map(molar_mass)
+    else:
+        iso = df.attrs["iso"]
+        molar_mass = molpar.get(id, iso, "mol_mass")
+
+        req = [molar_mass] * len(df.index)
```

Comment:
Are you sure you need to build a full list here? That's wasting memory. One float should be enough